File: ambeth/Ambeth.Merge/ambeth/util/ICacheHelper.cs
using System;
using System.Collections.Generic;
using De.Osthus.Ambeth.Merge.Model;

namespace De.Osthus.Ambeth.Util
{
    public interface ICacheHelper
    {
        [Obsolete]
        IPrefetchConfig CreatePrefetch();

        [Obsolete]
        IPrefetchState Prefetch(Object objects, IDictionary<Type, IList<String>> typeToPathToInitialize);
        
        Object CreateInstanceOfTargetExpectedType(Type expectedType, Type elementType);

        Object ConvertResultListToExpectedType(IList<Object> resultList, Type expectedType, Type elementType);

        Object[] ExtractPrimitives(IEntityMetaData metaData, Object obj);

        IObjRef[][] ExtractRelations(IEntityMetaData metaData, Object obj);

        IObjRef[][] ExtractRelations(IEntityMetaData metaData, Object obj, IList<Object> relationValues);
    }
}

File: jambeth/jAmbeth.Merge/src/de/osthus/ambeth/util/ICacheHelper.java
package de.osthus.ambeth.util;

import java.util.List;

import de.osthus.ambeth.collections.IMap;
import de.osthus.ambeth.merge.model.IEntityMetaData;
import de.osthus.ambeth.merge.model.IObjRef;

public interface ICacheHelper
{
	/**
	 * Please use <code>de.osthus.ambeth.util.IPrefetchHelper</code>
	 */
	@Deprecated
	IPrefetchConfig createPrefetch();

	/**
	 * Please use <code>de.osthus.ambeth.util.IPrefetchHelper</code>
	 */
	@Deprecated
	IPrefetchState prefetch(Object objects, IMap<Class<?>, List<String>> typeToMembersToInitialize);

	/**
	 * Please use <code>de.osthus.ambeth.util.IPrefetchHelper</code>
	 */
	@Deprecated
	IPrefetchState prefetch(Object objects);

	Object createInstanceOfTargetExpectedType(Class<?> expectedType, Class<?> elementType);

	Object convertResultListToExpectedType(List<Object> resultList, Class<?> expectedType, Class<?> elementType);

	Object[] extractPrimitives(IEntityMetaData metaData, Object obj);

	IObjRef[][] extractRelations(IEntityMetaData metaData, Object obj);

	/**
	 * 
	 * @param metaData
	 *            Meta data of obj
	 * @param obj
	 *            Entity to extract the relations from
	 * @param relationValues
	 *            Actual relation values (out param)
	 * @return ORI array for the relations
	 */
	IObjRef[][] extractRelations(IEntityMetaData metaData, Object obj, List<Object> relationValues);
}
